packages/block-editor/src/store/reducer.js

```diff
@@ -748,71 +741,92 @@ export function blockSelection( state = {
 			};
 		case 'MULTI_SELECT':
 			return {
-				...state,
-				start: action.start,
-				end: action.end,
-				initialPosition: null,
+				...BLOCK_SELECTION_INITIAL_STATE,
+				isMultiSelecting: state.isMultiSelecting,
+				start: { clientId: action.start },
+				end: { clientId: action.end },
 			};
 		case 'SELECT_BLOCK':
-			if ( action.clientId === state.start && action.clientId === state.end ) {
+			if (
+				action.clientId === state.start.clientId &&
+				action.clientId === state.end.clientId
+			) {
 				return state;
 			}
+
 			return {
-				...state,
-				start: action.clientId,
-				end: action.clientId,
+				...BLOCK_SELECTION_INITIAL_STATE,
 				initialPosition: action.initialPosition,
+				start: { clientId: action.clientId },
+				end: { clientId: action.clientId },
 			};
 		case 'REPLACE_INNER_BLOCKS': // REPLACE_INNER_BLOCKS and INSERT_BLOCKS should follow the same logic.
 		case 'INSERT_BLOCKS': {
 			if ( action.updateSelection ) {
 				return {
-					...state,
-					start: action.blocks[ 0 ].clientId,
-					end: action.blocks[ 0 ].clientId,
-					initialPosition: null,
-					isMultiSelecting: false,
+					...BLOCK_SELECTION_INITIAL_STATE,
+					start: { clientId: action.blocks[ 0 ].clientId },
+					end: { clientId: action.blocks[ 0 ].clientId },
 				};
 			}
+
 			return state;
 		}
 		case 'REMOVE_BLOCKS':
-			if ( ! action.clientIds || ! action.clientIds.length || action.clientIds.indexOf( state.start ) === -1 ) {
+			if (
+				! action.clientIds ||
+				! action.clientIds.length ||
+				action.clientIds.indexOf( state.start.clientId ) === -1
+			) {
 				return state;
 			}
-			return {
-				...state,
-				start: null,
-				end: null,
-				initialPosition: null,
-				isMultiSelecting: false,
-			};
-		case 'REPLACE_BLOCKS':
-			if ( action.clientIds.indexOf( state.start ) === -1 ) {
+
+			return BLOCK_SELECTION_INITIAL_STATE;
+		case 'REPLACE_BLOCKS': {
+			if ( action.clientIds.indexOf( state.start.clientId ) === -1 ) {
 				return state;
 			}
 
 			// If there are replacement blocks, assign last block as the next
 			// selected block, otherwise set to null.
 			const lastBlock = last( action.blocks );
-			const nextSelectedBlockClientId = lastBlock ? lastBlock.clientId : null;
 
-			if ( nextSelectedBlockClientId === state.start && nextSelectedBlockClientId === state.end ) {
+			if ( ! lastBlock ) {
+				return BLOCK_SELECTION_INITIAL_STATE;
+			}
+
+			if (
+				lastBlock.clientId === state.start.clientId &&
+				lastBlock.clientId === state.end.clientId
+			) {
 				return state;
 			}
 
 			return {
-				...state,
-				start: nextSelectedBlockClientId,
-				end: nextSelectedBlockClientId,
-				initialPosition: null,
-				isMultiSelecting: false,
+				...BLOCK_SELECTION_INITIAL_STATE,
+				start: { clientId: lastBlock.clientId },
+				end: { clientId: lastBlock.clientId },
 			};
+		}
 		case 'TOGGLE_SELECTION':
 			return {
-				...state,
+				...BLOCK_SELECTION_INITIAL_STATE,
```

Comment:
This change has the effect of deselecting the current block when toggling selection. Is that intended?

1. Insert an image block
2. Assign an image from the media library
3. Resize the image using the resizers

Expected: Block remains selected
Actual: Block is unselected

Reply:
I think I noticed something similar with the Spacer block as well when using the resize handle.